cogs/Pinner.py

```diff
@@ -0,0 +1,37 @@
+from re import L
+import traceback
+from discord.ext import commands
+from models.Pin import Pin
+from services import Embedder
+from util.Utility import Channels
+
+class Pinner(commands.Cog, name='Pinner', description='Used to pin a Message to the Pin Channel'):
+    factory_type = 'pin'
+
+    def __init__(self, bot):
+        self.bot = bot
+        self._embedder = Embedder.EmbedderFactory.get_embedder(Pinner.factory_type)
+
+    @commands.command(name='pin', description='Reply to a message with this command to pin it to the Pin channel', aliases=['this'])
+    async def pin(self, ctx: commands.Context):
+        try:
+            if not ctx.message.reference:
+                await ctx.message.channel.send('You have to reply .pin to the message you want pinned.')
+                return
+            
+            pin_channel = self.bot.get_channel(Channels.PINS.value)
```

Comment:
Broken by deletion of `util.Utility`. Honestly not sure the best way to handle this, seems like there's no way around hardcoding the channel name or ID somewhere.

You could search the guild's channel list for a channel called "pins" (e.g. `channel = next(filter(lambda x: x.name == "pins", ctx.guild.channels), None)`), but then if the channel name changed (for whatever reason) you'd have to change this line. You could create a constant `PIN_CHANNEL_NAME = "pins"` either in this file or in config (which is basically what you had before). That would be better since in the event of a change, you'd only have to change it in one spot in the code.

You could search for it by the channel ID instead of the name, so the channel name can change without breaking the code, but if the channel were deleted and re-added for some reason it would still break.

I think the only way we could do it without hardcoding would be to store a `PIN_CHANNEL_ID` constant in config somewhere, then add a command to set the pin channel, e.g. .set_pin_channel, which will basically execute `PIN_CHANNEL_ID = ctx.channel.id` and write it out to config. It's most likely overkill but it is robust and avoids any hardcoding which kinda gets me off.

Any of those options sound fine to me so I'll leave it up to you.

Reply:
Def agree with setting a constant in there, I'm on it. I'll see if I can do a bit of digging to find the best solution to make it dynamic but realistically I think of a solution that fits all use-cases. if we search by name and the name is changed for whatever reason, broke. If we search by id and we delete it and make a new one, broke. If we delete it and change the name, double broke. Just the nature of the beast, knowwhatimsayinmatelol

For now, I think any dynamic solution is beyond the scope of this PR so I'll set a constant and come back to it later.